tests/modeling/test_expressions.py

```diff
@@ -335,6 +338,41 @@ def test_greekify_model(pheno):
 def test_get_pk_parameters(load_model_for_test, testdata, model_path, kind, expected):
     model = load_model_for_test(testdata / model_path)
     assert set(get_pk_parameters(model, kind)) == set(expected)
+    assert 'KE0' not in get_pk_parameters(model)
+
+    pkpd_model = load_model_for_test(testdata / "nonmem" / "pheno_real.mod")
+    pkpd_model = add_effect_compartment(pkpd_model, "linear")
+    assert 'KE0' not in get_pk_parameters(pkpd_model)
+
+
+def test_get_pd_parameters(load_model_for_test, testdata):
+    model = load_model_for_test(testdata / "nonmem" / "pheno.mod")
+    pkpd_model_baseline = set_direct_effect(model, "baseline")
+    pkpd_effect_comp_model_baseline = add_effect_compartment(model, "baseline")
+    pkpd_model_linear = set_direct_effect(model, "linear")
+    pkpd_effect_comp_model_linear = add_effect_compartment(model, "linear")
+    pkpd_model_sigmoid = set_direct_effect(model, "sigmoid")
+    pkpd_effect_comp_model_sigmoid = add_effect_compartment(model, "sigmoid")
+
+    assert get_pd_parameters(pkpd_model_baseline) == ['E0']
+    assert get_pd_parameters(pkpd_effect_comp_model_baseline) == ['E0', 'KE0']
+    assert get_pd_parameters(pkpd_model_linear) == ['E0', 'S']
+    assert get_pd_parameters(pkpd_effect_comp_model_linear) == ['E0', 'KE0', 'S']
+    assert get_pd_parameters(pkpd_model_sigmoid) == ['EC_50', 'E_max', 'n']
+    assert get_pd_parameters(pkpd_effect_comp_model_sigmoid) == ['EC_50', 'E_max', 'KE0', 'n']
+    assert get_pd_parameters(model) == []
+    assert (
```

Comment:
This could test the same thing with `assert not` instead of `== set()`, and when you do `.intersection(arg)` if `arg` is e.g. a list it will be converted to a `set`, meaning you can write `set(get_pd_parameters(pkpd_model_baseline)).intersection(get_pk_parameters(pkpd_model_baseline))`. Final thing here is that you can also store the pd parameters in a variable so it doesn't need to call `get_pd_parameters` twice (and it will be a little easier to read I think).